accounts/static/accounts/js/onboarding_movies.js

```diff
@@ -0,0 +1,522 @@
+}
+
+/**
+ * This function reveals the tooltip along with adding the right styling
+ * for the tooltip buttons.
+ * @param {*} movie_id
+ */
+function revealDetails(movie_id) {
+  if (currentTooltiptext) {
+    currentTooltiptext.style.display = "none";
+  }
+  var tooltip = document.getElementById(movie_id);
+  updateButtons(movie_id);
+
+  currentTooltiptext = tooltip;
+  currentTooltiptext.style.display = "block";
+}
+
+/**
+ * This function updates the buttons with the state of the movie in the user's preferences.
+ * It is called in the revealDetails function (which is called when the user clicks on a movie)
+ * and whenever the user hovers over a movie
+ */
+function updateButtons(movie_id) {
+  intID = parseInt(movie_id);
+  if (movies_liked.has(intID)) {
+    document.getElementById(movie_id + "_upvote").style.backgroundColor =
+      likeGreen;
+  } else if (movies_disliked.has(intID)) {
+    document.getElementById(movie_id + "_dislike").style.backgroundColor =
+      "red";
+  }
+  if (movies_watched.has(intID)) {
+    document.getElementById(movie_id + "_seen").style.backgroundColor = "blue";
+  } else if (watchlist.has(intID)) {
+    document.getElementById(movie_id + "_watchlist").style.backgroundColor =
+      "cornflowerblue";
+  }
+  if (movies_rewatch.has(intID)) {
+    document.getElementById(movie_id + "_rewatch").style.backgroundColor =
+      rewatchOrange;
+  }
+  if (movies_blocked.has(intID)) {
+    document.getElementById(movie_id + "_exclude").style.backgroundColor =
+      excludeRed;
+  }
+}
+
+/**
+ *
+ * Movie-db related functions
+ *
+ */
+
+/* Repurposed code from landing_page/js/index.js */
+async function searchMovies(event) {
+  if (event.key != "Enter") {
+    return;
+  }
+
+  searchString = searchbar.value.trim();
+
+  if (searchString) {
+    const csrfToken = await getCSRFToken();
+    fetch("/api/search/movies", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "X-CSRFToken": csrfToken,
+      },
+
+      body: JSON.stringify({ search: searchString, send_all: true }),
+    })
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error("Network response was not ok");
+        }
+
+        return response.json();
+      })
+      .then((data) => {
+        movieContainer.innerHTML = "";
+        data.movies.forEach((movie) => {
+          var movieDiv = createMovieDiv(movie);
+          movieContainer.appendChild(movieDiv);
+        });
+      });
+  } else {
+    movieContainer.innerHTML = ""; //TODO create new div to add response message ("Getting random movies...")
+    fetchMovies(50);
+  }
+}
+
+/**
+ * Fetches random movies from the database and adds them to the movie container.
+ * @param {int} numMovies the number of movies to fetch
+ * @returns
+ */
+async function fetchMovies(numMovies = 35) {
+  // Prevent multiple fetches at once
+  if (isLoading) return;
+  isLoading = true;
+
+  try {
+    // Fetch movies from the API
```

Comment:
comments here are pretty redundant
and also, comments aren't a good 'newline' for readability, having a comment in b/w two lines of code is gross